### PR TITLE
changed the putcode to always be string and added a test

### LINKED ISSUE #46: Bug: putcode is sometimes stored as a number, sometimes a string
If there is no external identifier, the putcode is converted into a string for storage (https://github.com/adsabs/orcid-service/blob/master/orcid_service/views.py#L573). Otherwise, it's taken as-is from the ORCID API return - in this case, the type is a number (https://github.com/adsabs/orcid-service/blob/master/orcid_service/views.py#L612).

The fix should leave the types as numbers, and remove the string conversion.

### PR DESCRIPTION
**Pull request by femalves**

### CODE REVIEW

File: orcid_service/tests/test_service.py [hunk 1]
```diff
@@ -530,5 +533,8 @@ def request_callback(request, uri, headers):
         self.assertStatus(r, 200)
         self.assertTrue(r.json['name']['family-name']['value'] == 'Payne')
 
+    def assert_type_string(self, value):
+        assert isinstance(value, str), f"Value is not of type string. Found {type(value)} instead."
```
Comment (tjacovich) [inline orcid_service/tests/test_service.py]: The only tweak I would make would be to keep this in the `TestCase` framework ie.
```
  self.assertTrue(type(value), str)
```
or something along those lines in order to keep it consistent with the other tests.

**Review by tjacovich — CHANGES_REQUESTED** [review]
Just a quick tweak of the test to keep it consistent with the others and I think it's good to go.

File: orcid_service/tests/test_service.py (hunk 1 above)
Comment (femalves) [reply orcid_service/tests/test_service.py]: Done

**Review by tjacovich — APPROVED** [review]
Looks good to me!

**Pull request merged**